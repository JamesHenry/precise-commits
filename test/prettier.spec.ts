import { readFixtures } from './test-utils';
import {
  resolvePrettierConfigForFile,
  formatRangesWithinContents,
} from '../src/prettier';

const fixtures = readFixtures();

describe('prettier', () => {
  describe('resolvePrettierConfigForFile()', () => {
    fixtures.forEach(
      ({ fixtureName, fileExtension, initialContents, stagedContents }) => {
        it(fixtureName, () => {
          const prettierConfig = resolvePrettierConfigForFile(
            `./test/fixtures/${fixtureName}/initial${fileExtension}`,
          );
          expect(prettierConfig).toMatchSnapshot();
        });
      },
    );
  });

  describe('formatRangesWithinContents()', () => {
    it('should format the given ranges of the given source', () => {
      const contents = `
        var a = 1
        var b = 2
        var c = 3
      `;
      const formatted = formatRangesWithinContents(
        [{ rangeStart: 0, rangeEnd: 10 }, { rangeStart: 46, rangeEnd: 62 }],
        contents,
        {
          semi: true,
        },
      );
      expect(formatted).toEqual(`
        var a = 1;
        var b = 2
        var c = 3;
      `);
    });
  });
});
