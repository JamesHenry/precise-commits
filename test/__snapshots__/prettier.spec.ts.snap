// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier resolvePrettierConfigForFile() brand-new-jsx-file 1`] = `
Object {
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
}
`;

exports[`prettier resolvePrettierConfigForFile() empty-javascript-file 1`] = `
Object {
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
}
`;

exports[`prettier resolvePrettierConfigForFile() multiple-updated-lines-javascript-file 1`] = `
Object {
  "semi": true,
  "singleQuote": false,
  "trailingComma": "none",
}
`;

exports[`prettier resolvePrettierConfigForFile() single-updated-line-typescript 1`] = `
Object {
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
}
`;
